src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -396,33 +397,43 @@ void ComponentFiller::addConstraints(const LinearProblemApi::FillContext& ctx)
     }
 }
 
+void ComponentFiller::addStaticObjective(
+  const Optimization::TimeDependentLinearExpression& expression) const
+{
+    auto& pb = optimEntityContainer_.Problem();
+    const auto& solverVariables = optimEntityContainer_.getVariables();
+
+    for (const auto& [index, value]: expression[0])
+    {
+        pb.setObjectiveCoefficient(solverVariables[index].get(), value);
+    }
+}
+
 void ComponentFiller::addObjectives(const LinearProblemApi::FillContext& ctx)
 {
     auto* model = component_.getModel();
-    const auto& solverVariables = optimEntityContainer_.getVariables();
     ReadLinearExpressionVisitor visitor(optimEntityContainer_, ctx, component_);
 
     for (const auto& objective: model->Objectives() | locationFilter())
     {
+        const auto root_node = objective.expression().RootNode();
```

Comment:
```c++
const auto root_node = objective.expression().RootNode();
```
Now **root_node**  is defined, we should use it instead of `objective.expression().RootNode()` whenever it comes up.